src/inmanta/ast/__init__.py

```diff
@@ -644,7 +644,7 @@ class ExternalException(RuntimeException):
     it is wrapped in an ExternalException to make it conform to the expected interface
     """
 
-    def __init__(self, stmt: Locatable, msg: str, cause: Exception) -> None:
+    def __init__(self, stmt: "Optional[Locatable]", msg: str, cause: Exception) -> None:
```

Comment:
If I don't do this there are circular dependencies between the files/imports.

Reply:
I've rechecked them all:
this one is not needed so i'll remove it, but If I remove the others the issue comes back.

Reply:
It doesn't necessarily only occur when new imports are added, it might occur just as well when another module imports this one if it introduces a cycle. While previously the other module could be fully loaded so that this module can access its names, now the module is only partially loaded (because it's waiting for this one) so its names aren't available at load time yet.